src/main/java/fr/polytech/sim/log/NamedLogger.java

```diff
@@ -16,4 +16,19 @@ protected NamedLogger(String name) {
         this.name = name;
     }
 
+    /**
+     * Cette méthode hydrate le message avec les arguments passés et retourne un message formatté avec un message
+     * sur une/plusieurs lignes distinctes
+     *
+     * @param format le message à formatter
+     * @param args les arguments pour hydrater le message
+     * @return un message formatté avec les arguments hydratés sur une/plusieurs lignes distinctes
+     */
+    public void log(String format, Object... args) {
+        String entry = String.format(format, args);
+        String message=String.format("%s\t%s\n", this.name, entry);
+        this.writeLog(message);
+    }
+
+    public abstract void writeLog(String message);
```

Comment:
Cette méthode devrai être d'accès `protected`.
```suggestion
    protected abstract void writeLog(String message);
```